### ISSUE TITLE
loss explosion when training on custom Dataset

### ISSUE THREAD

**Issue by LokiXun** [MCG-NKU/E2FGVI#75]
Hi, it an awesome work! May I ask some help, I met some problems when training the model on REDS video dataset. When the training elapses about 40K iterations, the loss suddenly explode and the predict image became un-identifiable.  
![image](https://github.com/MCG-NKU/E2FGVI/assets/55100748/1b6b1567-8df7-4957-8021-68dbe5fb2eb6)
ps: the loss value show in picture is the sum of last 100 iterations

In order to run this dataset, I do the following modifications:
1. Dataset: the frame_size=1280x72 100 frames video. I crop them to 256x256 and add random blur. I use 7 local frames and 5 reference frames (which is equally sample from whole video except the local frame region). My objective is to deblur so i do not use the mask to cover the origin image
2. In order to train,  I modify the SoftSplit and Tansformer's parameter: `output_size = (64, 64)`
in [this line](https://github.com/MCG-NKU/E2FGVI/blob/709cbe319edc21b8a365a28e14cba595a93d62cf/model/e2fgvi_hq.py#L160C32-L160C32) and  `small_window_size = (11, 11)` to match the [12, 22, 22, 512] size feature out of Softsplit.
3. I set `no_dis: 1` in config file to not using the adversarial loss and gan_loss, I thought it may cause training unstable so I dismiss it
4. I only have one 24G-memory 4090 GPU so I could only set batchsize=1 and I did not change the scheduler which means the learning rate for the whole time is 1e-4.

the predict result at the loss-explosion iteration is like
![grid_164_39300_030_00000000](https://github.com/MCG-NKU/E2FGVI/assets/55100748/cc853d84-3fcd-4501-b16b-9651c79cdace)
ps: the first row: first 7 pic is local frames and latter 5 pic are non-local image; second row is correspinding GT. 3rd row is model's prediction

Does I mistakenly modified the param in TimeFocalTransformer? Have u guys have simiar issue and how u solve it, thanks.

**Comment by asfaukas** [comment]
Dear @LokiXun, have you solved this problem? The loss function increase at about 40k iterations.

**Comment by stayhungry1** [comment]
Is the loss increase from the DCN layer in the training

**Comment by Paper99** [comment]
Yes, the simple solution is resuming from a non-crashed checkpoint.

stayhungry1 ***@***.***> 于2024年4月9日周二 15:06写道：
